lib/Configuration.js

```diff
@@ -263,15 +221,39 @@ export default class Configuration {
         normalizePath(fromFile, this.workingDirectory),
         normalizePath(appliesFrom, this.workingDirectory)
       );
-    })[key];
+    });
+    return this.mergedValue(
+      applyingConfigs.map((config: Object): any => config[key]),
+      key,
+      fromFile
+    );
+  }
 
-    if (typeof value === 'function') {
-      return value({
-        pathToCurrentFile: this.pathToCurrentFile,
-        pathToImportedModule: fromFile,
-      });
+  mergedValue(values: Array<any>, key: string, fromFile: string): any {
+    let mergedResult;
+    for (let i = 0; i < values.length; i++) {
+      let value = values[i];
+      if (typeof value === 'function') {
+        value = value({
+          pathToCurrentFile: this.pathToCurrentFile,
+          pathToImportedModule: fromFile,
+        });
+      }
+      if (MERGABLE_CONFIG_OPTIONS.indexOf(key) === -1) {
```

Comment:
We could bring in a polyfill for this pretty easily. We could grab airbnb-js-shims and be set for a bunch of things, or we can bring it in directly from https://github.com/ljharb/array-includes